### DISCUSSION TITLE
Set preset temperatures at run time

### DISCUSSION THREAD

**Discussion by nagyrobi** [ScratMan/HASmartThermostat#14] [Ideas]
One of the problems of HA's built-in `generic_thermostat` is that the temperature for `away` mode is hardcoded in the configuration, it cannot be adjusted at run time.

Thankfully `smart_thermostat` supports `away`, `eco`, `boost`, `comfort`, `home`, `sleep`, `activity` each with its own temperature value configurable.

It would be nice to have the possibility to adjust these during runtime, and use HA's history database to always restore to the last known value after restart.

**Comment by ScratMan** [comment]
Hello,

Done in [release v2021.11.9](https://github.com/ScratMan/HASmartThermostat/releases/tag/2021.11.9)

**Reply by nagyrobi** [reply]
Thank you

**Comment by perhusaas** [comment]
This sound good, am I understanding it correct that if you adjust the temperature in run-time while being in preset "away", this temp level will be the new "away" preset level? (and same with all other presets)

the manual says "The temperature can then be adjusted using the set_preset_temp service, new value being restored after restarting HA." which indicates that the preset can only be adjusted by service?

**Reply by ScratMan** [reply]
> This sound good, am I understanding it correct that if you adjust the temperature in run-time while being in preset "away", this temp level will be the new "away" preset level? (and same with all other presets)

Yes


> the manual says "The temperature can then be adjusted using the set_preset_temp service, new value being restored after restarting HA." which indicates that the preset can only be adjusted by service?

Yes. But you can create an  input_number entity through UI ( https://my.home-assistant.io/redirect/helpers/ ) and add an automation that will be triggered when it's value is changed, calling the service with the input value as parameter.

**Reply by perhusaas** [reply]
I just now tested with this configuration: 

`  - platform: smart_thermostat`
`    name: Gj1 - STermostat`
`    unique_id: climate.gj1_stermostat`
`    heater: switch.varme_gjesterom_1_2`
`    target_sensor: sensor.viking_0203502038_d7_00_temperature`
`    min_temp: 8`
`    max_temp: 25`
`    ac_mode: false`
`    cold_tolerance: 0.5`
`    hot_tolerance: 0`
`    min_cycle_duration:`
`      seconds: 300`
`    keep_alive:`
`      minutes: 3`
`    initial_hvac_mode: "heat"`
`    precision: 0.1`
`    away_temp: 12`
`    comfort_temp: 20`
`    sleep_temp: 16`

when i'm in away mode, and I adjust the temperature (on the thermostat UI), the preset goes to "None", and the new value isn't stored for "away"?

**Reply by ScratMan** [reply]
The temperature knob on the thermostat doesn't change the presets setting. You must use the service to set them. Use this link to be redirected to the service development tool to set the desired temperature for the preset you want.
https://my.home-assistant.io/redirect/developer_call_service/?service=smart_thermostat.set_preset_temp

**Reply by perhusaas** [reply]
Yes, I've found the service call, no problem. 
What I don't understand is how there can be so many Virtual Thermostats for HASS (at least 5?), and none of them store a UI adjusted value to the preset setting? 
The VTs i've used in Domoticz, Vera, openLuup all do this, and I find it much more useful to be able to adjust these in the UI instead of going into YAML on a computer or through elaborate scripting..  Is it due to HASS users being so fond of scripting? who knows..

Yours is at least the best of them as you at least store the values through a restart. :) Thanks for creating a good VT!
